legend-engine-test-runner-service/src/main/java/org/finos/legend/engine/test/runner/service/ServiceTestRunner.java

```diff
@@ -334,6 +340,29 @@ private RichServiceTestResult executeTestAsserts(SingleExecutionPlan executionPl
         }
     }
 
+    private ConstantResult primitiveInstanceValueCollector(ValueSpecification instance)
+    {
+        return new ConstantResult(
+        instance instanceof CBoolean
+        ? ((CBoolean) instance).values.get(0)
+        : instance instanceof CString
+        ? ((CString)instance).values.get(0)
+        : instance instanceof CInteger
+        ? ((CInteger) instance).values.get(0)
+        : instance instanceof CFloat
+        ? ((CFloat) instance).values.get(0)
+        : instance instanceof CDateTime
+        ? ((CDateTime) instance).values.get(0)
+        : instance instanceof CStrictDate
+        ? ((CStrictDate) instance).values.get(0)
+        : instance instanceof CStrictTime
+        ? ((CStrictTime) instance).values.get(0)
+        : instance instanceof EnumValue
+        ? ((CString) instance).values.get(0)
+        : null
```

Comment:
Returning null can be risky here, because we might end run test with null when  user provided a value. Should we fail here instead?

Reply:
Changed to returning an exception